haystack/management/commands/update_index.py

```diff
@@ -30,11 +31,46 @@
     from datetime import datetime
     now = datetime.now
 
+from timeit import default_timer
 
 DEFAULT_BATCH_SIZE = None
 DEFAULT_AGE = None
 APP = 'app'
 MODEL = 'model'
+WORKER_TIMEOUT = 60 # seconds
```

Comment:
I think we should set this to unlimited by default.  This is a highly tunable setting but I think for most people they don't want it set in the simple case.